2003-02-13  Alan Modra  <amodra@bigpond.net.au>

	* write.c (TC_FORCE_RELOCATION_SUB_SAME): Revert last change.
	* config/tc-s390.h (TC_FORCE_RELOCATION_SUB_SAME): Define.

--- gas/write.c	23 Jan 2003 12:51:04 -0000	1.68
+++ gas/write.c	12 Feb 2003 22:56:38 -0000
@@ -51,7 +51,7 @@
 
 #ifndef TC_FORCE_RELOCATION_SUB_SAME
 #define TC_FORCE_RELOCATION_SUB_SAME(FIX, SEG)	\
-  (! SEG_NORMAL (SEG) || TC_FORCE_RELOCATION (FIX))
+  (! SEG_NORMAL (SEG))
 #endif
 
 #ifndef TC_FORCE_RELOCATION_SUB_ABS
--- gas/config/tc-s390.h	5 Sep 2002 00:01:18 -0000	1.9
+++ gas/config/tc-s390.h	12 Feb 2003 22:56:38 -0000
@@ -32,6 +32,10 @@ struct fix;
 #define TC_FORCE_RELOCATION(FIX) tc_s390_force_relocation(FIX)
 extern int tc_s390_force_relocation PARAMS ((struct fix *));
 
+/* Don't resolve foo@PLT-bar to offset@PLT.  */
+#define TC_FORCE_RELOCATION_SUB_SAME(FIX, SEG)	\
+  (! SEG_NORMAL (SEG) || TC_FORCE_RELOCATION (FIX))
+
 #define tc_fix_adjustable(X)  tc_s390_fix_adjustable(X)
 extern int tc_s390_fix_adjustable PARAMS ((struct fix *));
 
